src/validator/GoValidator.cpp

```diff
@@ -125,6 +126,44 @@ Status GoValidator::validateYield(YieldClause* yield) {
     return Status::OK();
 }
 
+Status GoValidator::validateTruncate(TruncateClause* truncate) {
+    if (truncate == nullptr) {
+        return Status::OK();
+    }
+    random_ = truncate->isSample();
+    auto* tExpr = truncate->truncate();
+    if (tExpr->kind() != Expression::Kind::kList) {
+        return Status::SemanticError("`%s' type must be LIST", tExpr->toString().c_str());
+    }
+    // lenght of list must be equal to N step
+    uint32_t totalSteps = steps_.mToN == nullptr ? steps_.steps : steps_.mToN->nSteps;
+    if (static_cast<const ListExpression*>(tExpr)->size() != totalSteps) {
+        return Status::SemanticError(
+            "`%s' length must be equal to %d", tExpr->toString().c_str(), totalSteps);
+    }
+    // check if value of list is non-integer
+    auto existNonInteger = [](const Expression* expr) -> bool {
+        if (expr->kind() != Expression::Kind::kConstant &&
+            expr->kind() != Expression::Kind::kList) {
+            return true;
+        }
+        if (expr->kind() == Expression::Kind::kConstant) {
+            auto& val = static_cast<const ConstantExpression*>(expr)->value();
+            if (!val.isInt()) {
+                return true;
+            }
+        }
+        return false;
+    };
+    FindVisitor visitor(existNonInteger);
+    tExpr->accept(&visitor);
+    auto res = visitor.results();
+    if (res.size() == 1) {
+        return Status::SemanticError("`%s' must be INT", res.front()->toString().c_str());
+    }
+    return Status::OK();
```

Comment:
We could return error for now since we don't implement it.

Reply:
ok